crates/autopilot/src/database/on_settlement_event_updater.rs

```diff
@@ -5,11 +5,11 @@ use {
     number_conversions::u256_to_big_decimal,
 };
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Default, Clone)]
 pub struct SettlementUpdate {
     pub block_number: i64,
     pub log_index: i64,
-    pub auction_id: i64,
+    pub auction_id: Option<i64>,
```

Comment:
Let's change this so that the required data is grouped together. We always need block_number, log_index, tx_from, tx_nonce.

The rest is needed all or nothing, so it should be something like:
```rust
struct AuctionData {
    pub auction_id: i64,
    pub gas_used: U256,
    pub effective_gas_price: U256,
    pub surplus: U256,
    pub fee: U256,
```

and then in `SettlementUpdate` `pub auction_data: Option<AuctionData>` .

Reply:
Done.